src/main/java/com/rtb/projectmanagementtool/user/UserSettingsServlet.java

```diff
@@ -0,0 +1,174 @@
+
+  public UserSettingsServlet() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    auth = new AuthOps(datastore);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+
+    // Authenticate
+    auth.loginUser(request, response);
+    Long userLoggedInId = auth.whichUserIsLoggedIn(request, response);
+    if (userLoggedInId == /*No user found*/ -1l) {
+      // If no user found, redirect to create user servlet
+      response.sendRedirect("/login");
+      return;
+    }
+
+    // HARDCODE FOR TESTING
+
+    // initialize/fill hardcoded userMajors
+    ArrayList<String> majors = new ArrayList<>();
+    majors.add("Chemistry");
+    majors.add("Studio Art");
+
+    // Create hardcoded user
+    long userID = Long.parseLong(request.getParameter("userID"));
+    String AuthID = "abc";
+    String userName = "Name1";
+    long userYear = 2023;
+    ArrayList<String> userMajors = majors;
+    Skills skills = Skills.OOP;
+    long userTotalCompTasks = 3;
+    UserData user =
+        new UserData(AuthID, userName, userYear, userMajors, skills, userTotalCompTasks);
+    user.setUserID(userID);
+
+    // NON TESTING: ONCE EVERYTHING IS SET UP
+
+    // new UserController
+    // UserController userController = new UserController(datastore);
+    // get user by ID long AuthOps auth = new AuthOps(datastore);
+    // long userID = auth.whichUserIsLoggedIn(request, response);
+    // user = userController.getUserByID(userID);
+
+    // make a string of majors
+    String majorsString = "";
+    for (String major : user.getUserMajors()) {
+      if (majorsString == "") {
+        majorsString = major;
+      } else {
+        majorsString = majorsString + "," + major;
+      }
+    }
+    majorsString = majorsString.replaceAll("\\s", "");
+
+    // make enum ArrayList<String>
+    // Skills userSkills = user.getUserSkills();
+    // Skills[] skillsArray = userSkills.values();
+    // ArrayList<String> skillsStringList = new ArrayList<>();
+    // for (Skills skill : skillsArray) {
+    // skillsStringList.add(skill.name());
+    // }
+
+    // pre checked buttons
+    ArrayList<String> checkedStatus = new ArrayList<>(Arrays.asList("", "", "", "", "", "", ""));
+    Skills getSkills = user.getUserSkills();
+    String skillsAsString = getSkills.name();
+    if (skillsAsString.equals("NONE")) {
+      checkedStatus.set(0, "checked");
+    } else if (skillsAsString.equals("LEADERSHIP")) {
+      checkedStatus.set(1, "checked");
+    } else if (skillsAsString.equals("ORGANIZATION")) {
+      checkedStatus.set(2, "checked");
+    } else if (skillsAsString.equals("WRITING")) {
+      checkedStatus.set(3, "checked");
+    } else if (skillsAsString.equals("ART")) {
+      checkedStatus.set(4, "checked");
+    } else if (skillsAsString.equals("WEBDEV")) {
+      checkedStatus.set(5, "checked");
+    } else {
+      checkedStatus.set(6, "checked");
+    }
```

Comment:
Can a switch-case be used?

Reply:
I'll look into it. Good suggestion, thanks!